tests/unit/query-cache-test.js

```diff
@@ -922,4 +922,52 @@ module('unit/query-cache', function(hooks) {
       assert.equal(this.adapterAjax.called, false, '`adapter.ajax` is not called');
     });
   });
+
+  test('.cacheURL inserts consistently inserts a promise into cache', function(assert) {
+    assert.expect(2);
+    const mockResult = { id: 1, foo: 'bar' };
+
+    this.queryCache.cacheURL('foo-bar', mockResult);
+    const cachedResult = this.queryCache.queryURL('/foo-bar', { cacheKey: 'foo-bar' });
+
+    assert.ok(cachedResult && cachedResult.then, 'cachedResult is a promise');
+
+    return cachedResult.then(result => {
+      assert.deepEqual(mockResult, result, 'queryURL returned cached result');
+    });
+  });
+
+  test('.cacheURL supports unloading of models and invalidates cache', function(assert) {
+    assert.expect(2);
+
+    const firstPayload = {
```

Comment:
I prefer `let` for locally-scoped vars like this unless we're actually `Object.freeze`ing

that said I don't care much, but we should at least be consistent between `firstPayload` and `secondPayload` as neither appears to be reassigned.